### PR TITLE
MODQM-447: Auto-populate LCCN 010 $a value with spaces

### PR DESCRIPTION
**Pull request by mukhiddin-yusuf**
### Purpose
_Modify the space population Regex patterns for LCCN a$ value_

### Approach
- add space between prefix and digits when it is not of enough length for the prefix according to the Structure A&B

### Changes Checklist
- [ ] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [ ] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [ ] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MODQM-447](https://folio-org.atlassian.net/browse/MODQM-447)

### CODE REVIEW

File: src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java [hunk 1]
```diff
@@ -22,10 +22,12 @@
 public class Tag010FieldItemPopulationService extends FieldItemMarcPopulationService {
 
   private static final Pattern LCCN_FIELD_PATTERN = Pattern.compile("[$][abz].*$");
-  private static final Pattern LCCN_10_L_FIELD_PATTERN = Pattern.compile("\\d{10}");
-  private static final Pattern D_8 = Pattern.compile("\\d{8}");
+  private static final Pattern LCCN_10_DIGITS_ONLY_PATTERN = Pattern.compile("\\d{10}");
+  private static final Pattern LCCN_10_ONLY_LETTER_PREFIX_PATTERN = Pattern.compile("[a-z]\\d{10}");
+  private static final Pattern LCCN_8_DIGITS_ONLY_PATTERN = Pattern.compile("\\d{8}");
+  private static final Pattern LCCN_8_ONE_OR_TWO_LETTER_PREFIX_PATTERN = Pattern.compile("([a-z][ |a-z]? ?)(\\d{8})");
```
Comment (viacheslavkol) [inline src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java:28]: Considering mod-record-specifications PR I would assume that pipe `|` is not a valid character, is it?

**Review by viacheslavkol — APPROVED** [review]

File: src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java [hunk 2]
```diff
@@ -48,14 +50,26 @@ public boolean supportFormat(MarcFormat marcFormat) {
...
-        lccnString = lccnString.concat(SPACE);
+      } else {
+        var matcher = LCCN_8_ONE_OR_TWO_LETTER_PREFIX_PATTERN.matcher(lccnString);
+        if (matcher.matches()) {
+          var prefix = matcher.group(1).trim();
+          var digits = matcher.group(2);
+          lccnString = prefix.concat(repeat(SPACE_CHARACTER, LCCN_OLD_PREFIX_LENGTH - prefix.length()))
+            .concat(digits).concat(SPACE);
+        }
       }
```
Comment (psmagin) [inline src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java:73]: Could you separate logic that is related to structure a and structure b?

File: src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java [hunk 3]
```diff
@@ -22,10 +22,12 @@
...
 
   private static final Pattern LCCN_FIELD_PATTERN = Pattern.compile("[$][abz].*$");
-  private static final Pattern LCCN_10_L_FIELD_PATTERN = Pattern.compile("\\d{10}");
-  private static final Pattern D_8 = Pattern.compile("\\d{8}");
+  private static final Pattern LCCN_10_DIGITS_ONLY_PATTERN = Pattern.compile("\\d{10}");
+  private static final Pattern LCCN_10_ONLY_LETTER_PREFIX_PATTERN = Pattern.compile("[a-z]\\d{10}");
+  private static final Pattern LCCN_8_DIGITS_ONLY_PATTERN = Pattern.compile("\\d{8}");
+  private static final Pattern LCCN_8_ONE_OR_TWO_LETTER_PREFIX_PATTERN = Pattern.compile("([a-z][ a-z]? ?)(\\d{8})");
+  private static final Pattern LCCN_8_NO_TRAILING_SPACE_PATTERN = Pattern.compile("[a-z]{3}\\d{8}");
   private static final Pattern D_8_S_DOLLAR_SIGN = Pattern.compile("\\d{8}\\s/.*$");
```
[the remark this answers is not in the document]
Comment (mukhiddin-yusuf) [reply src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java:30]: I was also not sure, is there use case at all, and thus did not touch on it and was thinking maybe someone provides one :-)
OK, will leave and rename it for now)

File: src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java [hunk 4]
```diff
@@ -48,14 +50,26 @@ public boolean supportFormat(MarcFormat marcFormat) {
   private Subfield subfieldFromString(String string) {
     String lccnString = string.substring(2).trim();
     if (LCCN_FIELD_PATTERN.matcher(string).matches()) {
```
[the remark this answers is not in the document]
Comment (mukhiddin-yusuf) [reply src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java:52]: Indeed, right, was left from previous converter's code, will get rid of it

File: src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (psmagin) [reply src/main/java/org/folio/qm/service/population/impl/Tag010FieldItemPopulationService.java]: Let's remove it as it's not clear an will provide no match for just numbers lccn

**Review by psmagin — APPROVED** [review]

**Pull request merged**